client/keys/types.go

```diff
@@ -19,13 +30,26 @@ type RecoverKey struct {
 	Index    int    `json:"index,string,omitempty"`
 }
 
+// NewRecoverKey constructs a new RecoverKey request structure.
+func NewRecoverKey(password, mnemonic string, account, index int) RecoverKey {
+	return RecoverKey{Password: password, Mnemonic: mnemonic, Account: account, Index: index}
+}
+
 // UpdateKeyReq requests updating a key
 type UpdateKeyReq struct {
 	OldPassword string `json:"old_password"`
 	NewPassword string `json:"new_password"`
 }
 
+// NewUpdateKeyReq constructs a new UpdateKeyReq structure.
+func NewUpdateKeyReq(old, new string) UpdateKeyReq {
+	return UpdateKeyReq{OldPassword: old, NewPassword: new}
+}
+
 // DeleteKeyReq requests deleting a key
 type DeleteKeyReq struct {
 	Password string `json:"password"`
 }
+
+// NewDeleteKeyReq constructs a new DeleteKeyReq structure.
+func NewDeleteKeyReq(password string) DeleteKeyReq { return DeleteKeyReq{Password: password} }
```

Comment:
In go usually anything `New*` returns a pointer to the struct though, no? Meaning return `&DeleteKeyReq{}` here?

Reply:
> In go usually anything New* returns a pointer to the struct though, no?

I used to believe so. Apparently the use of the `New` prefix is idiomatic regardless of whether the function returns a struct or a pointer.